### PR TITLE
WIP Iam get policy

### PR DESCRIPTION
**Pull request by josenrihernand**
Hi all,

This is a WIP PR to implement IAM get_policy:

I include the following code, in order to implement "get_policy" method from IAM (I have not modified ..moto/iam/models.py file).

https://github.com/josenrihernand/moto/blob/master/tests/test_iam/test_iam.py#L268#L284
https://github.com/josenrihernand/moto/blob/IAM_get_policy/moto/iam/responses.py#L181#L186
https://github.com/josenrihernand/moto/blob/IAM_get_policy/moto/iam/responses.py#L846#L858

As from the test (of the get_policy method) on my local environment, I can see that I get the value:

groupPolicies = iam_conn_client.list_attached_group_policies(
GroupName = groups["GroupName"]
for attachedPolicy in groupPolicies["AttachedPolicies"]:
currentPolicy = iam_conn_client.get_policy(         ---> get_policy method works.
PolicyArn = attachedPolicy["PolicyArn"]
);

print("Attached Policy: ", attachedPolicy["PolicyArn"]); --> it works.
--> ('Attached Policy: ', 'arn:aws:iam::aws:policy/TestDBPolicy')

print("Current Policy: ", currentPolicy);	--> it also works.
--> ('Current Policy: ', {u'Policy': {u'CreateDate': datetime.datetime(2017, 10, 4, 15, 45, 35, tzinfo=tzutc()), u'DefaultVersionId': ''}, 'ResponseMetadata': {'RetryAttempts': 0, 'HTTPStatusCode': 200, 'RequestId': '20f7279f-99ee-11e1-a4c3-27EXAMPLE804', 'HTTPHeaders': {u'Content-Type': u'text/plain', u'server': u'amazon.com'}}})

However, when I run the method "get_policy_version", it fails with the following error message:

groupPolicies = iam_conn_client.list_attached_group_policies(
GroupName = groups["GroupName"]
policy_version = iam_conn_client.get_policy_version(   ----> get_policy_version fail.
PolicyArn = attachedPolicy["PolicyArn"],
VersionId = currentPolicy["Policy"]["DefaultVersionId"]
);
An error occurred (NoSuchEntity) when calling the GetPolicyVersion operation: Policy version not found

I think the cause is because the get_policy method (I implemented it on my local environment) does not return "DefaultVersionId" value.

So, I  would like to ask you some help on this. Specifically, I don't sure if the method "get_policy" was implemented well. 

Could you please check / review / test the code I included to implement get_policy ?
Could you please verify why DefaultVersionId does not return any value (and so the error Policy version not found) ?

I really would appreciate you help ! since I need to implement get_policy on moto library!

Thanks,

### CODE REVIEW

**Comment by josenrihernand** [comment]
In the previous input, instead of:
https://github.com/josenrihernand/moto/blob/master/tests/test_iam/test_iam.py#L268#L284
I meant:
https://github.com/josenrihernand/moto/blob/IAM_get_policy/tests/test_iam/test_iam.py#L286#L301

thanks,

File: moto/iam/responses.py [hunk 1]
```diff
@@ -20,6 +20,20 @@ def detach_role_policy(self):
         template = self.response_template(GENERIC_EMPTY_TEMPLATE)
         return template.render(name="DetachRolePolicyResponse")
 
+    def attach_group_policy(self):
```
Comment (JackDanger) [inline moto/iam/responses.py]: If you rebase your work on the latest Moto master branch you'll see that these already exist. Great minds think alike! :)

Try this:
```
git remote set-url origin https://github.com/spulec/moto.git
git remote set-url josenrihernand git@github.com:josenrihernand/moto.git
git fetch origin
git rebase origin/master
# handle merge conflicts
git push -f josenrihernand HEAD:IAM_get_policy
```

**Comment by josenrihernand** [comment]
@JackDanger , thanks for take a look on it!  
I'm aware that probably you're enough busy, so, I really appreciate you valuable time to take a look into it. 

I executed the git commands provided, and got these results / output:

[user@ip-xxx-xx-xx-xxx moto]# pwd
/root/moto-repo/moto

[user@ip-xxx-xx-xx-xxx moto]# git status
# On branch IAM_get_policy  (---> This is the branch i created for this issue)
nothing to commit, working directory clean

[user@ip-xxx-xx-xx-xxx moto]# git remote set-url origin https://github.com/spulec/moto.git
[user@ip-xxx-xx-xx-xxx moto]# echo $?
0 ---> this git command works (git remote set-url origin...).

[user@ip-xxx-xx-xx-xxx moto]# git remote set-url josenrihernand git@github.com:josenrihernand/moto.git
fatal: No such remote 'josenrihernand' ---> I got this error message.

After the error message, I executed "git remote -v", and I got this output:

[user@ip-xxx-xx-xx-xxx moto]# git remote -v
origin  https://github.com/spulec/moto.git (fetch)
origin  https://github.com/spulec/moto.git (push)

I did not continue with the next one (git getch origin), because the error message. What should i execute ?

thanks a lot!

**Comment by josenrihernand** [comment]
Hi @JackDanger 

Regarding the previous update, I  executed this:


[user@ip-xxx-xx-xx-xxx moto]# git remote set-url origin https://github.com/spulec/moto.git

[user@ip-xxx-xx-xx-xxx moto]# git remote set-url josenrihernand git@github.com:josenrihernand/moto.git
fatal: No such remote 'josenrihernand'

[user@ip-xxx-xx-xx-xxx moto]# git remote set-url josenrihernand git@github.com:josenrihernand/moto.git

[user@ip-xxx-xx-xx-xxx moto]# git remote add josenrihernand git@github.com:josenrihernand/moto.git

[user@ip-xxx-xx-xx-xxx moto]# git remote -v
josenrihernand  git@github.com:josenrihernand/moto.git (fetch)
josenrihernand  git@github.com:josenrihernand/moto.git (push)
origin  https://github.com/spulec/moto.git (fetch)
origin  https://github.com/spulec/moto.git (push)

[user@ip-xxx-xx-xx-xxx moto]# git fetch origin
remote: Counting objects: 1413, done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 1413 (delta 508), reused 511 (delta 492), pack-reused 884
Receiving objects: 100% (1413/1413), 470.94 KiB | 0 bytes/s, done.
Resolving deltas: 100% (993/993), completed with 142 local objects.
From https://github.com/spulec/moto
   f052757..371bb2b  master     -> origin/master
From https://github.com/spulec/moto
 * [new tag]         1.1.10     -> 1.1.10
 * [new tag]         1.1.11     -> 1.1.11
 * [new tag]         1.1.12     -> 1.1.12
 * [new tag]         1.1.13     -> 1.1.13
 * [new tag]         1.1.18     -> 1.1.18
 * [new tag]         1.1.19     -> 1.1.19
 * [new tag]         1.1.2      -> 1.1.2
 * [new tag]         1.1.20     -> 1.1.20
 * [new tag]         1.1.21     -> 1.1.21
 * [new tag]         1.1.3      -> 1.1.3
 * [new tag]         1.1.4      -> 1.1.4
 * [new tag]         1.1.5      -> 1.1.5
 * [new tag]         1.1.6      -> 1.1.6
 * [new tag]         1.1.8      -> 1.1.8
 * [new tag]         1.1.9      -> 1.1.9

[user@ip-xxx-xx-xx-xxx moto]# git rebase origin/master
First, rewinding head to replay your work on top of it...
Applying: New function get_policy and the corresponding template
Using index info to reconstruct a base tree...
M       moto/iam/responses.py
<stdin>:49: trailing whitespace.
warning: 1 line adds whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging moto/iam/responses.py
Applying: New function test_get_policy
Using index info to reconstruct a base tree...
M       tests/test_iam/test_iam.py
Falling back to patching base and 3-way merge...
Auto-merging tests/test_iam/test_iam.py


[user@ip-xxx-xx-xx-xxx moto]# git push -f josenrihernand HEAD:IAM_get_policy
Counting objects: 1569, done.
Compressing objects: 100% (505/505), done.
Writing objects: 100% (1423/1423), 367.43 KiB | 0 bytes/s, done.
Total 1423 (delta 1006), reused 1299 (delta 901)
remote: Resolving deltas: 100% (1006/1006), completed with 105 local objects.
To git@github.com:josenrihernand/moto.git
 + af8a9f6...57af5dc HEAD -> IAM_get_policy (forced update)


I'll appreciate your feedback,

thanks a lot !

**Comment by spulec** [comment]
Closed with https://github.com/spulec/moto/pull/1759

**Pull request closed without merging**